### PR TITLE
Support service implementation

### PR DESCRIPTION
**Pull request by KKarlash**
@bonewell 
@DKlimenko 
@VladSemenyuk 
@thinhts
@giangtqh

### CODE REVIEW

File: src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * @brief SupportServiceRequest command class
+ */
+class SupportServiceRequest {
+  //TODO(KKarlash): public BaseCommandGPBRequest
+ public:
+  /**
+   * @brief SupportServiceRequest class constructor
+   * @param parent pointer to VRModule
+   * @param message Message from mobile
+   **/
```
Comment (giangtqh) [inline src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h]: */

File: src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+ private:
+  vr_hmi_api::ServiceMessage message_;
+};
+
+}  // namespace commands
+
+}  // namespace vr_cooperation
+
+#endif /* SRC_COMPONENTS_VR_COOPERATION_INCLUDE_VR_COOPERATION_COMMANDS_SUPPORT_SERVICE_REQUEST_H_ */
```
Comment (giangtqh) [inline src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h]: //

**Comment by giangtqh** [comment]
reviewed

File: src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+namespace vr_cooperation {
+
+namespace commands {
+
+/**
+ * @brief SupportServiceRequest command class
+ */
+class SupportServiceRequest {
+  //TODO(KKarlash): public BaseCommandGPBRequest
```
Comment (thinhts) [inline src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h]: It's BaseCommandGPBRequest or RequestFromHMI
@giangtqh implement RequestFromHMI. Please discuss with him about class name

File: src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+ public:
+  /**
+   * @brief SupportServiceRequest class constructor
+   * @param parent pointer to VRModule
+   * @param message Message from mobile
+   **/
+   SupportServiceRequest(VRModule* parent);
```
Comment (thinhts) [inline src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h]: explicit

File: src/components/vr_cooperation/include/vr_cooperation/vr_module.h [hunk 5]
```diff
@@ -153,6 +159,21 @@ class VRModule : public functional_modules::GenericModule,
...
+   * @param app_id default vr-service app id
+   */
+  void set_supported(bool supported) {
+    supported_ = supported;
+  }
+
+  /**
+   * @brief Returns unique correlation ID for next HMI request
+   */
+  uint32_t GetNextCorrelationID() {
```
Comment (thinhts) [inline src/components/vr_cooperation/include/vr_cooperation/vr_module.h]: const function

File: src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h (hunk 2 above)
Comment (bonewell) [inline src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h]: If you use Eclipse you can set up define guards.

**Comment by thinhts** [comment]
Reviewed

**Comment by DKlimenko** [comment]
reviewed

File: src/components/vr_cooperation/src/support_service_request.cc [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+void SupportServiceRequest::Execute() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  message_.set_rpc(vr_hmi_api::SUPPORT_SERVICE);
+  message_.set_rpc_type(vr_hmi_api::REQUEST);
+  //TODO(KKarlash): uncomment after HMI factory implementation
+  //message_.set_correlation_id(parent_->GetNextCorrelationID());
+  //parent_->SendMessageToHMI();
```
Comment (bonewell) [inline src/components/vr_cooperation/src/support_service_request.cc]: you have `void VRModule::SendMessageToHMI(const vr_hmi_api::ServiceMessage& message);`
Why did you comment these lines? You have both methods in VRModule.

File: src/components/vr_cooperation/src/support_service_request.cc [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+void SupportServiceRequest::OnEvent(/*
+    const event_engine::Event<vr_hmi_api::ServiceMessage,
+    vr_hmi_api::RPCName>& event*/) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  //TODO(KKarlash): Uncomment after HMI factory implementation
```
Comment (bonewell) [inline src/components/vr_cooperation/src/support_service_request.cc]: I don't understand why you can not implement this without HMI factory.

**Comment by bonewell** [comment]
Reviewed.

File: src/components/vr_cooperation/src/support_service_request.cc (hunk 6 above)
Comment (KKarlash) [inline src/components/vr_cooperation/src/support_service_request.cc]: parent_ will be in base class

File: src/components/vr_cooperation/src/support_service_request.cc (hunk 7 above)
Comment (KKarlash) [inline src/components/vr_cooperation/src/support_service_request.cc]: there is other event engine will be implemented by Giang

File: src/components/vr_cooperation/include/vr_cooperation/vr_module.h (hunk 5 above)
Comment (KKarlash) [inline src/components/vr_cooperation/include/vr_cooperation/vr_module.h]: couldn't be const

File: src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h (hunk 4 above)
Comment (KKarlash) [inline src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h]: will have 2 parameters

File: src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h (hunk 3 above)
Comment (KKarlash) [inline src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h]: will be updated after his merge

File: src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h (hunk 2 above)
Comment (KKarlash) [inline src/components/vr_cooperation/include/vr_cooperation/commands/support_service_request.h]: got you, thanks

**Pull request merged**